app/main.py

```diff
@@ -1,3 +1,83 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+from pathlib import Path
+
+from app.car import Car
+from app.customer import Customer
+from app.shop import Shop
+
+
+def shop_trip() -> None:
+    path_to_file = Path(__file__).parent.joinpath("config.json")
+    with open(path_to_file, "r") as file_read_stream:
+        file_data = json.load(file_read_stream)
+
+    fuel_price = file_data["FUEL_PRICE"]
+    customers = []
+    shops = []
+    for i in range(len(file_data["customers"])):
+        customers.append(
+            Customer(
+                file_data["customers"][i]["name"],
+                file_data["customers"][i]["product_cart"],
+                file_data["customers"][i]["location"],
+                file_data["customers"][i]["money"],
+                Car(
+                    file_data["customers"][i]["car"]["brand"],
+                    file_data["customers"][i]["car"]["fuel_consumption"]
+                ),
+            )
+        )
+
+    for i in range(len(file_data["shops"])):
+        shops.append(
+            Shop(
+                file_data["shops"][i]["name"],
+                file_data["shops"][i]["location"],
+                file_data["shops"][i]["products"]
+            )
+        )
+
+    for customer in customers:
+        print(f"{customer.name} has {customer.money} dollars")
+
+        best_trip = customer.calculate_fuel(
+            customer.location,
+            shops[0].location
+        ) * fuel_price + customer.count_purchases(shops[0].products)
+        best_shop_name = shops[0].name
```

Comment:
firstly calculate all necessary values
secondly print what u need
P.S.
print better to place into separated func

Reply:
Done